components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/RegistryPropertiesResource.java

```diff
@@ -0,0 +1,489 @@
+                jsonBody.put("message", "Successfully added the registry property");
+            } else {
+                jsonBody = Utils.createJsonError("Invalid payload for properties", axis2MessageContext, BAD_REQUEST);
+            }
+        } else {
+            jsonBody = Utils.createJsonError("Error while fetching properties from payload", axis2MessageContext,
+                    BAD_REQUEST);
+        }
+        return jsonBody;
+    }
+
+    /**
+     * This method returns a string containing properties.
+     *
+     * @param axis2MessageContext   AXIS2 message context
+     * @return                      A string containing file content
+     */
+    public static String getPayload(org.apache.axis2.context.MessageContext axis2MessageContext) {
+
+        try {
+            InputStream jsonStream = JsonUtil.getJsonPayload(axis2MessageContext);
+            return IOUtils.toString(jsonStream);
+        } catch (Exception e) {
+            return null;
+        }
+    }
+
+    /**
+     * This method fetches properties to add/modify.
+     *
+     * @param content   Content of the properties file
+     * @return          Fetched properties
+     */
+    private Properties getProperties(String content) {
+        Properties properties = new Properties();
+        try {
+            JSONArray jsonArray = new JSONArray(content);
+            for (int i = 0; i < jsonArray.length(); i++) {
+                JSONObject jsonObject = jsonArray.getJSONObject(i);
+                properties.put(jsonObject.get(NAME), jsonObject.get(VALUE_KEY));
+            }
+        } catch (JSONException e) {
+            return null;
+        }
+        return properties;
+    }
+
+    /**
+     * This method adds and modifies properties.
+     *
+     * @param oldProperties Existing properties
+     * @param properties    Properties to be added
+     * @return              New properties
+     */
+    private Properties getNewProperties(Properties oldProperties, Properties properties) {
+
+        for (Object property : properties.keySet()) {
+            Object value = properties.get(property);
+            oldProperties.put(property, value);
+        }
+        return oldProperties;
+    }
+
+    /**
+     * This method handles DELETE.
+     *
+     * @param messageContext      Synapse message context
+     * @param axis2MessageContext AXIS2 message context
+     */
+    private void handleDelete(MessageContext messageContext,
+            org.apache.axis2.context.MessageContext axis2MessageContext) {
+
+        String registryPath = Utils.getQueryParameter(messageContext, REGISTRY_PATH);
+        String propertyName = Utils.getQueryParameter(messageContext, NAME);
+        String validatedPath;
+        axis2MessageContext.removeProperty(Constants.NO_ENTITY_BODY);
+        JSONObject jsonBody;
+        if (Objects.nonNull(propertyName)) {
```

Comment:
This pattern in repeated over and over again. Try to intoduce a util method to do all these validations. We can pass the jsonBody to that method so that it can change the outgoing error message

Reply:
Changed it to a util method